app/queues/dns/create-record-worker.server.ts

```diff
@@ -0,0 +1,145 @@
+  async (job) => {
+    logger.info('Creating a record in DB');
+    const { username, type, name, value } = job.data;
+
+    const existingRecords = await prisma.record.count({
+      where: {
+        type,
+        name,
+        value,
+      },
+    });
+
+    if (existingRecords) {
+      throw new Error('Record already exists');
+    }
+
+    try {
+      const result = await prisma.record.create({
+        data: {
+          username,
+          type,
+          name,
+          value,
+          expiresAt: setMonthsFromNow(6),
+          status: RecordStatus.pending,
+        },
+      });
+      return result.id as Record['id'];
+    } catch (error) {
+      logger.warn('Could not create a record in DB', error);
+      throw error;
+    }
+  },
+  { connection: redis }
+);
+
+export const createDnsRecordWorker = new Worker<JobRecord>(
+  createDnsRecordQueueName,
+  async (job) => {
+    logger.info('Creating a record in Route 53');
+    const { username, type, name, value } = job.data;
+
+    try {
+      return createRecord(username, type, name.toLowerCase(), value);
+    } catch (error) {
+      logger.warn('Could not create a record in Route53', error);
+      throw error;
+    }
+  },
+  { connection: redis }
+);
+
+export const checkDnsStatusWorker = new Worker(
+  checkDnsStatusQueueName,
+  async (job) => {
+    logger.info('Checking Route53 readiness');
+
+    const childJobValues: { [jobKey: string]: string } = await job.getChildrenValues();
+    const createDnsRecordJobKey = Object.keys(childJobValues).find((item) =>
+      item.includes(createDnsRecordQueueName)
+    );
+
+    if (!createDnsRecordJobKey) {
+      throw new Error(`Could not receive data from child job ${createDnsRecordQueueName}`);
+    }
+    const changeId = childJobValues[createDnsRecordJobKey];
+
+    const status = await getChangeStatus(changeId);
+
+    if (status === 'PENDING') {
+      throw new Error('Change status is still pending');
+    }
+    return status;
+  },
+  { connection: redis }
+);
+
+export const syncDbStatusWorker = new Worker(
+  syncDbStatusQueueName,
+  async (job) => {
+    logger.info('Sync DB status with DNS');
+
+    const childJobValues = await job.getChildrenValues();
+    const keysForChildValues = Object.keys(childJobValues);
+    if (keysForChildValues.length !== 2) {
+      throw new UnrecoverableError('Invalid children values from previous workers');
+    }
+
+    const addDbRecordJobKey = keysForChildValues.find((item) =>
+      item.includes(addDbRecordQueueName)
+    );
+    const checkStatusJobKey = keysForChildValues.find((item) =>
+      item.includes(checkDnsStatusQueueName)
+    );
+
+    if (!addDbRecordJobKey) {
+      throw new Error(`Could not receive data from child job ${addDbRecordQueueName}`);
+    }
+    if (!checkStatusJobKey) {
+      throw new Error(`Could not receive data from child job ${checkDnsStatusQueueName}`);
+    }
+    const recordId = childJobValues[addDbRecordJobKey] as Record['id'];
```

Comment:
```suggestion
    const recordId = childJobValues[addDbRecordJobKey] as JobRecordResult;
```

This way you can be 100% sure that what you return and what you get are always in sync type-wise, and if you change the interface in the `add db record` worker, it wil also be changed here too

Same thing goes with `status job`, I just didn't want to repeat the example